src/main/scala-2.12/com/linagora/gatling/imap/protocol/ImapProtocol.scala

```diff
@@ -2,33 +2,31 @@ package com.linagora.gatling.imap.protocol
 
 import java.util.Properties
 
-import akka.actor.{ActorRef, ActorSystem}
+import akka.actor.ActorRef
 import com.linagora.gatling.imap.protocol.Command.Disconnect
 import io.gatling.core.CoreComponents
 import io.gatling.core.config.GatlingConfiguration
 import io.gatling.core.protocol.{Protocol, ProtocolComponents, ProtocolKey}
 import io.gatling.core.session.Session
 
 object ImapProtocol {
-  val ImapProtocolKey = new ProtocolKey {
-    override type Protocol = ImapProtocol
-    override type Components = ImapComponents
+  val ImapProtocolKey = new ProtocolKey[ImapProtocol, ImapComponents] {
 
     override def protocolClass: Class[io.gatling.core.protocol.Protocol] = classOf[ImapProtocol].asInstanceOf[Class[io.gatling.core.protocol.Protocol]]
 
-    override def defaultValue(configuration: GatlingConfiguration): ImapProtocol = throw new IllegalStateException("Can't provide a default value for ImapProtocol")
+    override def defaultProtocolValue(configuration: GatlingConfiguration): ImapProtocol = throw new IllegalStateException("Can't provide a default value for ImapProtocol")
 
-    override def newComponents(system: ActorSystem, coreComponents: CoreComponents): ImapProtocol => ImapComponents = { protocol =>
-      val sessions: ActorRef = system.actorOf(ImapSessions.props(protocol), "imapsessions")
+    override def newComponents(coreComponents: CoreComponents): ImapProtocol => ImapComponents = { protocol =>
+      val sessions: ActorRef = coreComponents.actorSystem.actorOf(ImapSessions.props(protocol), "imapsessions")
       ImapComponents(protocol, sessions)
     }
   }
 }
 
 case class ImapComponents(protocol: ImapProtocol, sessions: ActorRef) extends ProtocolComponents {
-  override def onStart: Option[(Session) => Session] = None
+  override def onStart: Session => Session = s => s
```

Comment:
Is there identity function in Scala?

Reply:
I didn't know until now but you have one in scala.Predef.identity

Reply:
It's more expressive to use it, and avoid handcrafted expressions